### PR TITLE
Upgrade Golang to match upstream Gardener

### PR DESCRIPTION
**Pull request by briantopping**
Signed-off-by: Brian Topping <brian.topping@sap.com>

**Release note**:

```other developer
Upgrade Golang to 1.18.4 to match upstream Gardener
```

### CODE REVIEW

File: .test-defs/provider-vsphere.yaml [hunk 1]
```diff
@@ -12,4 +12,4 @@ spec:
       go run -mod=vendor ./test/tm/generator.go
       --infrastructure-provider-config-filepath=$INFRASTRUCTURE_PROVIDER_CONFIG_FILEPATH
       --controlplane-provider-config-filepath=$CONTROLPLANE_PROVIDER_CONFIG_FILEPATH
-  image: golang:1.17.11
+  image: golang:1.18.4
```
Comment (MartinWeindel) [inline .test-defs/provider-vsphere.yaml:15]: let's use the latest version 
```suggestion
  image: golang:1.18.5
```

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ############# builder
-FROM golang:1.17.11 AS builder
+FROM golang:1.18.4 AS builder
```
Comment (MartinWeindel) [inline Dockerfile:2]:
```suggestion
FROM golang:1.18.5 AS builder
```

**Review by MartinWeindel — CHANGES_REQUESTED** [review]
minor version of golang has already been increased again.
I will also fix the verify problem on the meta branch

**Comment by MartinWeindel** [comment]
@briantopping You need to run `make revendor`

**Comment by briantopping** [comment]
Yeap, I saw that… thanks 😊

From: Martin Weindel ***@***.***>
Date: Thursday, August 4, 2022 at 8:57 AM
To: gardener/gardener-extension-provider-vsphere ***@***.***>
Cc: Topping, Brian ***@***.***>, Mention ***@***.***>
Subject: Re: [gardener/gardener-extension-provider-vsphere] Upgrade Golang to match upstream Gardener (PR #288)

@briantopping<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fbriantopping&data=05%7C01%7Cbrian.topping%40sap.com%7C0b94659fdf23465d1e2808da7629a6f5%7C42f7676cf455423c82f6dc2d99791af7%7C0%7C0%7C637952218524134733%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=%2BkIrFdXwKIqrEOjQWCm7fHtsmZ%2BmqAJhO46uq6g3mSw%3D&reserved=0> You need to run make revendor

**Review by MartinWeindel — APPROVED** [review]

**Pull request merged**